src/detect-mqtt-connect-flags.c

```diff
@@ -134,30 +134,36 @@ static int DetectMQTTConnectFlagsMatch(DetectEngineThreadCtx *det_ctx,
 static DetectMQTTConnectFlagsData *DetectMQTTConnectFlagsParse(const char *rawstr)
 {
     DetectMQTTConnectFlagsData *de = NULL;
+    char *copy = NULL;
     int ret = 0;
     int ov[MAX_SUBSTRINGS];
 
     ret = DetectParsePcreExec(&parse_regex, rawstr, 0, 0, ov, MAX_SUBSTRINGS);
     if (ret < 1) {
         SCLogError(SC_ERR_PCRE_MATCH, "invalid flag definition: %s", rawstr);
-        return NULL;
+        goto error;
     }
 
     de = SCCalloc(1, sizeof(DetectMQTTConnectFlagsData));
     if (unlikely(de == NULL))
-        return NULL;
+        goto error;
     de->username = de->password = de->will = MQTT_DONT_CARE;
     de->will_retain = de->clean_session = MQTT_DONT_CARE;
 
-    char copy[strlen(rawstr)+1];
-    strlcpy(copy, rawstr, sizeof(copy));
+    copy = SCStrdup(rawstr);
```

Comment:
I don't really understand the goal here. Why is the stack copy not working?

Reply:
It is working. I wanted to move the code path to use `goto error` everywhere to make sure that Coverity sees the complete lifetime of `de`. However, the compiler won't allow us to `goto` into a scope with a variably sized declaration (`copy`). So I chose to manage the memory manually here.